branch-3.4 -- bugfix -- ZOOKEEPER-2891

xoiss opened the pull request:
Fixes https://issues.apache.org/jira/browse/ZOOKEEPER-2891

phunt commented:
Hi @xoiss please update this PR to follow the guidelines in "generate a pull request" here: https://cwiki.apache.org/confluence/display/ZOOKEEPER/HowToContribute - in particular wrt the summary and description text which really helps folks identifying which PRs to prioritize. Thanks!

phunt commented:
@xoiss I like this patch but it's missing tests. Can you include the tests from https://issues.apache.org/jira/secure/attachment/12671215/ZOOKEEPER-1636.patch and resubmit?

phunt commented:
ps. You accidentally included the patch for 2890 in this PR, perhaps you can fix that at the same time? Thx.

anmolnar commented:
Closing because @xoiss is not responding for a while. Please re-open if you can come back to this.

xoiss commented:
// xoiss is here

hi, all!
unfortunately i don't have enough time to respond promptly
and moreover i don't check the gmail mail box because of... some reason

ok, what we can do with these?

anmolnar commented:
@xoiss This one is still outstanding. Please take a look and rebase.

anmolnar commented:
ping @xoiss

xoiss commented:
Hello!
I have reworked the issue description, and also I have reworked this PR.

The improved issue: https://issues.apache.org/jira/browse/ZOOKEEPER-2891
I have tried to make the description more clear.
I'm not sure if I have succeeded, but I hope it became better.

The renewed PR: https://github.com/apache/zookeeper/pull/999
I have rebased it on the latest `branch-3.4`.
As I can see, it passed all precommit checks.

To **phunt**:
>  I like this patch but it's missing tests. Can you include the tests from https://issues.apache.org/jira/secure/attachment/12671215/ZOOKEEPER-1636.patch and resubmit?

I'm not sure if I understand what should I do.
I've checked the mentioned patch, and I see that it updates one of unit-tests - `TestMulti.cc`
This test is currently in the repo, and my changes don't contradict with it.
Moreover, it even does not have such a setup to reproduce the case which is patched with my fixes.
So, my question: what shall I do with it?

Thank you!
Alex (aka xoiss)

Please, revise my renewed PR: https://github.com/apache/zookeeper/pull/999

xoiss commented:
pong @anmolnar 
;]

anmolnar commented:
Closed, because duplicate of #999
